### PR TITLE
feat(filter): add facade API and docs

### PR DESCRIPTION
**Pull request by wzhudev**
close #xxx

## Pull Request Checklist

- [ ] Related tickets or issues have been linked in the PR description (or missing issue).
- [ ] [Naming convention](https://github.com/dream-num/univer/blob/dev/docs/NAMING_CONVENTION.md) is followed (**do please** check it especially when you created new plugins, commands and resources).
- [ ] Unit tests have been added for the changes (if applicable).
- [ ] Breaking changes have been documented (or no breaking changes introduced in this PR).

### CODE REVIEW

File: packages/facade/src/apis/sheets/f-filter.ts [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+        private readonly _filterModel: FilterModel,
+        @Inject(Injector) private readonly _injector: Injector,
+        @ICommandService private readonly _commandSrv: ICommandService
+    ) {
+        // empty
+    }
+
+    /**
+     * Get the filter criteria of a column.
+     * @param col The column number.
```
Comment (VicKun4937) [inline packages/facade/src/apis/sheets/f-filter.ts:41]:
```suggestion
     * @param {number} col The column number.
     
```
if the param is optional, should  like below:
```
 * @param {number} [col] The column number.
```

File: packages/facade/src/apis/sheets/f-filter.ts [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+     */
+    getColumnFilterCriteria(col: number): Nullable<IFilterColumn> {
+        return this._filterModel.getFilterColumn(col)?.getColumnData();
+    }
+
+    /**
+     * Clear the filter criteria of a column.
+     * @param col The column number.
+     * @returns If the filter criteria is cleared.
```
Comment (VicKun4937) [inline packages/facade/src/apis/sheets/f-filter.ts:50]: returns should give a type
```suggestion
     * @returns {boolean} If the filter criteria is cleared.
```

File: packages/facade/src/apis/sheets/f-range.ts [hunk 3]
```diff
@@ -658,4 +665,50 @@ export class FRange {
...
+
+    // #region Filter
+
+    /**
+     * Create a filter for the current range. If the worksheet already has a filter, this method would return `null`.
+     *
+     * @returns The interface class to handle the filter. If the worksheet already has a filter,
+     * this method would return `null`.
+     */
+    async createFilter(): Promise<FFilter | null> {
```
Comment (VicKun4937) [inline packages/facade/src/apis/sheets/f-range.ts:679]: https://jsdoc.app/tags-async

File: packages/sheets-filter-ui/src/commands/commands/sheets-filter.command.ts [hunk 4]
```diff
@@ -14,15 +14,90 @@
...
+                undoMutations: [{ id: RemoveSheetsFilterMutation.id, params: { unitId, subUnitId } }],
+                redoMutations: [redoMutation],
+            });
+        }
+
+        return result;
+    },
+};
+
+export const RemoveSheetFilterCommand: ICommand<ISheetCommandSharedParams> = {
```
Comment (VicKun4937) [inline packages/sheets-filter-ui/src/commands/commands/sheets-filter.command.ts:69]: here can use typedef , but it not must case

File: packages/facade/src/apis/sheets/f-range.ts [hunk 5]
```diff
@@ -658,4 +665,50 @@ export class FRange {
...
+        if (!success) return null;
+
+        return this.getFilter();
+    }
+
+    /**
+     * Get the filter for the current range's worksheet.
+     *
+     * @returns The interface class to handle the filter. If the worksheet does not have a filter,
+     * this method would return `null`.
```
Comment (VicKun4937) [inline packages/facade/src/apis/sheets/f-range.ts:697]: @return

File: packages/facade/src/apis/sheets/f-filter.ts (hunk 1 above)
Comment (wzhudev) [reply packages/facade/src/apis/sheets/f-filter.ts:41]: Got it.

File: packages/sheets-filter-ui/src/commands/commands/sheets-filter.command.ts (hunk 4 above)
Comment (wzhudev) [reply packages/sheets-filter-ui/src/commands/commands/sheets-filter.command.ts:69]: Could you please give an example? @VicKun4937

**Pull request merged**